emcc.py

```diff
@@ -1628,6 +1628,9 @@ def phase_setup(options, state, newargs):
       exit_with_error('SUPPORT_LONGJMP=wasm cannot be used with DISABLE_EXCEPTION_CATCHING=0')
     default_setting('DISABLE_EXCEPTION_THROWING', 1)
 
+  if user_settings.get('EXPORT_EXCEPTION_HANDLING_HELPERS') == '1' and (settings.DISABLE_EXCEPTION_CATCHING and not settings.WASM_EXCEPTIONS):
```

Comment:
In this case you can just do `if settings.EXPORT_EXCEPTION_HANDLING_HELPERS and ..`

(i.e. in this case we don't specifically care if it was the user who set it)

Reply:
Also do we now have any other way to detect exceptions being enabled other than checking two different settings (`settings.DISABLE_EXCEPTION_CATCHING and not settings.WASM_EXCEPTIONS`).

Reply:
> In this case you can just do `if settings.EXPORT_EXCEPTION_HANDLING_HELPERS and ..`
> 
> (i.e. in this case we don't specifically care if it was the user who set it)

Done

Reply:
> In this case you can just do `if settings.EXPORT_EXCEPTION_HANDLING_HELPERS and ..`
> 
> (i.e. in this case we don't specifically care if it was the user who set it)

Changing this errors out with:
```console
AssertionError: internal error: attempt to read setting 'EXPORT_EXCEPTION_HANDLING_HELPERS' while in limited settings mode
```

I guess that's because this method is `phase_setup`, which takes care of compile time setup, and `EXPORT_EXCEPTION_HANDLING_HELPERS` is not among [`COMPILE_TIME_SETTINGS`](https://github.com/emscripten-core/emscripten/blob/37e502a8eb8800322e1e984f32a2b24b783fbac8/tools/settings.py#L55-L83). We can add it to `COMPILE_TIME_SETTINGS` but exporting functions isn't really compile time settings. Or we can move these two lines to `phase_linker_setup`, but I kind of want to check EH-related things in one place. WDYT?

Reply:
I think the probably belongs where `EXPORT_EXCEPTION_HANDLING_HELPERS` is actually implemented in `phase_linker_setup` on line 2836, no?

Reply:
Moved the code to `phase_linker_setup` and tidied up again the whole relationship between `ASSERTIONS`, `EXCEPTION_STACK_TRACES` and `EXPORT_EXCEPTION_HANDLING_HELPERS`: https://github.com/emscripten-core/emscripten/pull/18696#issuecomment-1426579890